server/channels/app/syncables.go

```diff
@@ -272,9 +293,6 @@ func (a *App) SyncRolesAndMembership(rctx request.CTX, syncableID string, syncab
 		if err := a.deleteGroupConstrainedTeamMemberships(rctx, &syncableID); err != nil {
 			rctx.Logger().Warn("Error deleting group constrained team memberships", mlog.Err(err))
 		}
-		if err := a.ClearTeamMembersCache(syncableID); err != nil {
-			rctx.Logger().Warn("Error clearing team members cache", mlog.Err(err))
-		}
```

Comment:
I don't find a reason why clearing the user cache of a team member is necessary any longer. `SyncSyncableRoles` clears the cache of the users whose roles have changed.

Please double-check if I missed anything here that would make clearing the cache necessary.

Reply:
When a user is added to a team due to a group membership, `App.AddTeamMember` gets called. That later clears the session cache: https://github.com/mattermost/mattermost/blob/42c5af86f71dbf1b5eaf0fde52746d347159669a/server/channels/app/team.go#L795-L797